### PR TITLE
Minimal implementation of the array facade

### PR DESCRIPTION
**Pull request by JohanMabille**

### CODE REVIEW

File: src/array.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// limitations under the License.
+
+#include "sparrow/array.hpp"
+#include "sparrow/array_factory.hpp"
+#include "sparrow/layout/array_helper.hpp"
+
+namespace sparrow
+{
+    array::array(ArrowArray&& array, ArrowSchema&& schema)
+        : p_array(array_factory(arrow_proxy(std::move(array), std::move(schema))))
```
Comment (Alex-PLACET) [inline src/array.cpp:22]:
```suggestion
        : p_array(array_factory(arrow_proxy(std::forward<ArrowArray>(array), std::forward<ArrowSchemay>(schema))))
```

File: src/array.cpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    {
+    }
+
+    array::array(ArrowArray&& array, ArrowSchema* schema)
+        : p_array(array_factory(arrow_proxy(std::move(array), schema)))
```
Comment (Alex-PLACET) [inline src/array.cpp:27]:
```suggestion
        : p_array(array_factory(arrow_proxy(std::forward<ArrowArray>(array), schema)))
```

**Review by Alex-PLACET — APPROVED** [review]
LGTM

File: src/array.cpp (hunk 1 above)
Comment (DerThorsten) [reply src/array.cpp:22]: `std::move` is correct here.
`std::forward` is for generic code, but here are not templates

File: src/array.cpp (hunk 2 above)
Comment (DerThorsten) [reply src/array.cpp:27]: `std::move` is correct here.
`std::forward` is for generic code, but here are not templates

**Review by DerThorsten — APPROVED** [review]

**Pull request merged**